### PR TITLE
[JENKINS-58483, JENKINS-43669] - Remove Trilead SSH dependencies for CLI and the Core

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-58483](https://issues.jenkins-ci.org/browse/JENKINS-58483). This patch removes the explicit dependencies on Trilead SSH which lead to bundling of the library in the Jenkins WAR. even after 2.185

#4110 is probably a right solution in longer term, but it should hotfix the issue along with https://github.com/jenkinsci/ssh-cli-auth-module/pull/10 and https://github.com/jenkinsci/ssh-cli-auth-module/pull/14

### Proposed changelog entries

* JENKINS-58483 - Major bug: Remove Trilead SSH dependencies from the Jenkins core. These dependencies caused SSH build agent connection issues in 2.185
* JENKINS-43669 - RFE: Upgrade SSH CLI Auth Module from 1.5 to 1.7 to remove Trilead SSH references 
  * Changelog for 1.7: https://github.com/jenkinsci/ssh-cli-auth-module/releases/tag/ssh-cli-auth-1.7
  * Changelog for 1.6: https://github.com/jenkinsci/ssh-cli-auth-module/releases/tag/ssh-cli-auth-1.6
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -116,11 +116,6 @@ THE SOFTWARE.
       <artifactId>jnr-posix</artifactId>
       <version>3.0.45</version>
     </dependency>
-    <dependency>
-      <groupId>org.kohsuke</groupId>
-      <artifactId>trilead-putty-extension</artifactId>
```
Comment (oleg-nenashev) [inline core/pom.xml]: Used by several plugins in https://github.com/search?q=org%3Ajenkinsci+PuTTYKey&type=Code , but it should be addressed by Trilead API plugin

**Review by jglick — APPROVED** [review]
So this is amending #4085 IIUC?

The regression sounded pretty basic—SSH agents do not connect. Presumably this is covered by lots of ATH tests. Were none of those in the smoke test category for core? Is someone watching (full) ATH results for regressions?

**Comment by jglick** [comment]
And setting aside automated tests, can I assume that the issue is reproducible without the patch and not reproducible with the patch? Sorry for asking, but apparently #4085 was not manually tested (or at least not in a sufficiently realistic mode).

**Review by kuisathaverat — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Ideally we should add SSH Slaves Plugin to the testing pipeline once https://github.com/jenkinsci/jenkins/pull/3438 is finalized

File: cli/pom.xml [hunk 2]
```diff
@@ -77,11 +77,6 @@
       <artifactId>slf4j-jdk14</artifactId>
       <optional>true</optional> <!-- ditto -->
     </dependency>
-    <dependency> <!-- TODO remove and replace PrivateKeyProvider with SecurityUtils.createFileKeyPairProvider() as in SshClient -->
```
Comment (oleg-nenashev) [inline cli/pom.xml]: FTR it was done by @kuisathaverat in https://github.com/jenkinsci/jenkins/pull/3902

**Comment by oleg-nenashev** [comment]
It is going to fail: https://ci.jenkins.io/job/Core/job/jenkins/job/PR-4111/2/testReport/hudson.cli/CLITest/Linux_jdk8___Linux_Publishing___strictHostKey/ . 

```
org.apache.sshd.common.SshException: No more authentication methods available
	at org.apache.sshd.client.session.ClientUserAuthService.tryNext(ClientUserAuthService.java:314)
	at org.apache.sshd.client.session.ClientUserAuthService.processUserAuth(ClientUserAuthService.java:251)
	at org.apache.sshd.client.session.ClientUserAuthService.process(ClientUserAuthService.java:198)
	at org.apache.sshd.common.session.helpers.AbstractSession.doHandleMessage(AbstractSession.java:614)
	at org.apache.sshd.common.session.helpers.AbstractSession.handleMessage(AbstractSession.java:547)
	at org.apache.sshd.common.session.helpers.AbstractSession.decode(AbstractSession.java:1498)
	at org.apache.sshd.common.session.helpers.AbstractSession.messageReceived(AbstractSession.java:508)
	at org.apache.sshd.common.session.helpers.AbstractSessionIoHandler.messageReceived(AbstractSessionIoHandler.java:66)
	at org.apache.sshd.common.io.nio2.Nio2Session.handleReadCycleCompletion(Nio2Session.java:301)
	at org.apache.sshd.common.io.nio2.Nio2Session$1.onCompleted(Nio2Session.java:281)
	at org.apache.sshd.common.io.nio2.Nio2Session$1.onCompleted(Nio2Session.java:278)
	at org.apache.sshd.common.io.nio2.Nio2CompletionHandler.lambda$completed$0(Nio2CompletionHandler.java:38)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.apache.sshd.common.io.nio2.Nio2CompletionHandler.completed(Nio2CompletionHandler.java:37)
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126)
	at sun.nio.ch.Invoker$2.run(Invoker.java:218)
	at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Clearly we have some test automation for CLI @jglick  🎉

**Review by fcojfernandez — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Some updates here:

* I reproduced the issue manually and confirmed that the proposed patch resolves the issue
* I tried using test automation in https://github.com/jenkinsci/ssh-slaves-plugin/pull/135 . Apparently the issue does not show up in JTH, likely due to https://issues.jenkins-ci.org/browse/JENKINS-41827
* I checked the generated WARs. Trilead files are not longer included there
* I launched https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/plugins/SshSlavesPluginTest.java . It passes with the patch (and with installed Trilead API which was less trivial)

I believe this change is ready to go

**Review by daniel-beck — APPROVED** [review]
🤷‍♂

**Pull request merged**